README.md

```diff
@@ -0,0 +1,74 @@
+A bunch of hacks making [weeder](https://github.com/ndmitchell/weeder) work for projects built with [haskell.nix](https://github.com/input-output-hk/haskell.nix/)
+
+### Usage example
+
+Suppose you have a project with three packages in a layout like this (note that package name and directory name may differ):
+
+```
+super-project
+|-- stack.yaml
+|-- super-package.cabal
+|-- …
+|-- prelude
+    |-- super-prelude.cabal
+    |-- …
+|-- super-tool
+    |-- super-tool.cabal
+    |-- …
+```
+
+Import this library, providing nixpkgs package set:
+
+```nix
+weeder-hacks = import (builtins.fetchTarball https://github.com/serokell/haskell-nix-weeder/archive/master.tar.gz) (import <nixpkgs> {});
+```
+
+In the call to `haskell-nix.stackProject` add options for each package to generate .dump-hi files used by weeder and save them to the build output:
+
+```nix
```

Comment:
It's weird how haskell.nix returns both local packages and dependencies in a single set, and AFAIU it not trivial to apply a piece of configuration to local packages only: https://github.com/input-output-hk/haskell.nix/issues/298. And in this case it is an example configuration, so we go with the most straightforward code possible, the user can implement a way to avoid repetetion themselves.